src/numerics/petsc_vector.C

```diff
@@ -390,7 +390,7 @@ void PetscVector<T>::add (const T a_in, const NumericVector<T> & v_in)
   libmesh_assert_equal_to (this->size(), v->size());
 
   VecGhostLocalFormRAII<T> ghost(*this, this->type() == GHOSTED);
-  VecGhostLocalFormRAII<T> v_ghost(*v, this->type() == GHOSTED);
+  VecGhostLocalFormRAII<T> v_ghost(*v, v->type() == GHOSTED);
```

Comment:
There are several other places where the same issue may exist. Could you fix other places as well?